firefox-ios/Storage/Rust/RustLogins.swift

```diff
@@ -709,6 +738,23 @@ public class RustLogins: LoginsProtocol {
         return deferred
     }
 
+    public func listLogins(completionHandler: @escaping (Result<[EncryptedLogin], Error>) -> Void) {
```

Comment:
FYI `listLogins` is added in https://github.com/mozilla-mobile/firefox-ios/pull/19206, maybe we should only have `searchLoginsWithQuery` in this PR? Considering the task and title. Want to avoid conflicts as much as possible :)

Reply:
Once that PR is merged, I will happily delete **listLogins** from here. I added this method here only to see if it works and also to have this PR opened. Thank you 🙏🏻